### PR TITLE
RBN-96: Fetching one book

### PR DESCRIPTION
**Pull request by isidora-b**
### What did you implement?

Implemented fetching book by id functionality, added custom exception and exception handler, added unit tests and integration tests.

### How can we verify the changes?

Changes can be verified by running the app and checking '/books/id' endpoint with Swagger and by running integration and unit tests.

### More info

- Jira: [RBN-96](https://productdock.atlassian.net/browse/RBN-96)

### CODE REVIEW

File: rbc/src/main/java/com/library/rbc/controller/BookController.java [hunk 1]
```diff
@@ -23,4 +26,13 @@ public Page<BookDto> getAllBooks(
       @ParameterObject @PageableDefault(size = 12) Pageable pageable) {
     return bookService.getAllBooks(pageable);
   }
+
+  @GetMapping("/{id}")
+  public ResponseEntity<BookDto> getBook(@PathVariable String id) {
+    BookDto bookDto = bookService.getBook(id);
+    if (bookDto != null) {
```
Comment (teodoraBr) [inline rbc/src/main/java/com/library/rbc/controller/BookController.java]: Is there a case where bookDto could be null?

File: rbc/src/main/java/com/library/rbc/controller/BookController.java [hunk 2]
```diff
@@ -23,4 +26,13 @@ public Page<BookDto> getAllBooks(
       @ParameterObject @PageableDefault(size = 12) Pageable pageable) {
     return bookService.getAllBooks(pageable);
   }
+
+  @GetMapping("/{id}")
+  public ResponseEntity<BookDto> getBook(@PathVariable String id) {
```
Comment (teodoraBr) [inline rbc/src/main/java/com/library/rbc/controller/BookController.java]: You can simplify this by returning just BookDto.

File: rbc/src/test/java/com/library/rbc/controller/bookcontroller/BookControllerShould.java [hunk 3]
```diff
@@ -38,4 +43,24 @@ void getAllBooks() {
...
+    when(bookService.getBook(BOOK_ID)).thenReturn(bookDto);
+    ResponseEntity<BookDto> result = bookController.getBook(BOOK_ID);
+
+    ResponseEntity<BookDto> expected = createResponseEntity();
+    assertEquals(expected, result);
+  }
+
+  @Test
+  void getResponseWhenNoBookIsFound() {
+    when(bookService.getBook(BOOK_ID)).thenReturn(null);
```
Comment (teodoraBr) [inline rbc/src/test/java/com/library/rbc/controller/bookcontroller/BookControllerShould.java]: When book by given id is not found we expect to throw a BookNotFoundException

File: rbc/src/test/java/com/library/rbc/controller/bookcontroller/BookControllerShould.java [hunk 4]
```diff
@@ -38,4 +43,29 @@ void getAllBooks() {
...
   }
+
+  @Test
+  void getBook() {
+    BookDto bookDto = createBookDto();
+
+    when(bookService.getBook(BOOK_ID)).thenReturn(bookDto);
+    BookDto result = bookController.getBook(BOOK_ID);
+
+    BookDto expected = createBookDto();
```
Comment (teodoraBr) [inline rbc/src/test/java/com/library/rbc/controller/bookcontroller/BookControllerShould.java]: This line is not needed. You can assert that the result is equal to bookDto.

File: rbc/src/test/java/com/library/rbc/service/bookservice/BookServiceShould.java [hunk 5]
```diff
@@ -48,4 +56,31 @@ void getAllBooks() {
...
+  @Test
+  void getBook() {
+    Book book = createBook();
+    BookDto bookDto = createBookDto();
+
+    when(bookRepository.findById(BOOK_ID)).thenReturn(Optional.of(book));
+    when(bookMapper.bookToBookDto(book)).thenReturn(bookDto);
+    BookDto result = bookService.getBook(BOOK_ID);
+
+    BookDto expectedBookDto = createBookDto();
```
Comment (teodoraBr) [inline rbc/src/test/java/com/library/rbc/service/bookservice/BookServiceShould.java]: Same comment as for controller test.

**Review by teodoraBr — DISMISSED** [review]

**Review by Aleksandra-Romanic — APPROVED** [review]

**Review by teodoraBr — APPROVED** [review]

**Pull request merged**